app/workers/deletes.php

```diff
@@ -92,6 +95,25 @@ public function shutdown(): void
     {
     }
 
+    /**
+     * @param Document $document teams document
+     * @param string $projectId
+     */
+    protected function deleteCollection(Document $document, string $projectId): void
+    {
+        $collectionId = $document->getId();
+
+        $dbForInternal = $this->getInternalDB($projectId);
```

Comment:
Don't we need to cleanup anything from external DB? or it's cleaned automatically?

Reply:
The `$dbForExternal->deleteCollection($collectionId)` call in the Database controller handles two responsibilities:

- [Drops the whole table from MariaDB](https://github.com/utopia-php/database/blob/fc6304cae24d344a3277812c89b6923c9f12582c/src/Database/Adapter/MariaDB.php#L178-L185)
- [Cleans up the document from the `metadata` collection](https://github.com/utopia-php/database/blob/fc6304cae24d344a3277812c89b6923c9f12582c/src/Database/Database.php#L350-L355)

There shouldn't be anything more to clean up from `$dbForExternal` after that, so we can use the Deletes worker to remove the associated documents from `$dbForInternal` in tables: `collections`, `attributes`, and `indexes`.

Thank you for the review @lohanidamodar - while answering your question, I realized that we didn't call `$dbForInternal->deleteDocument('collections', $collectionId)`. Added in bfdf6de